encodings/dict/benches/dict_compress.rs

```diff
@@ -38,10 +38,17 @@ fn decode(c: &mut Criterion) {
 
     let mut rng = StdRng::seed_from_u64(334);
 
-    let primitive_arr = gen_primitive_for_dict::<i32>(&mut rng, 1_000_000, 50);
+    let primitive_arr = gen_primitive_for_dict::<u8>(&mut rng, 1_000_000, 32);
+    let dict = dict_encode(primitive_arr.as_ref()).unwrap();
+    group.throughput(Throughput::Bytes(primitive_arr.nbytes() as u64));
+    group.bench_function("dict_decode_primitives_u8", |b| {
+        b.iter_with_setup(|| dict.clone(), |dict| dict.into_canonical().unwrap())
+    });
+
+    let primitive_arr = gen_primitive_for_dict::<i32>(&mut rng, 1_000_000, 32);
     let dict = dict_encode(primitive_arr.as_ref()).unwrap();
     group.throughput(Throughput::Bytes(primitive_arr.nbytes() as u64));
-    group.bench_function("dict_decode_primitives", |b| {
+    group.bench_function("dict_decode_primitives_i32", |b| {
```

Comment:
Would be nice to run the bench over many different params.

I guess 8k upto 64k (len/codes) and maybe between 4 and 1024k unqiue values?

Reply:
Can change this in a follow up PR but would then also move the benchmark to divan.

Reply:
@joseph-isaacs Will expand in a follow up here.